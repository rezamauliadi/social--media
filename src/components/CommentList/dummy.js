export default [
  {
    postId: 1,
    id: 1,
    name: "id labore",
    email: "Eliseo@gardner.biz",
    body: "laudantium "
  },
  {
    postId: 1,
    id: 2,
    name: "quo vero ",
    email: "Jayne_Kuhic@sydney.com",
    body: "est natus enim nihil est "
  },
  {
    postId: 1,
    id: 3,
    name: "odio adipisci",
    email: "Nikita@garfield.biz",
    body: "quia molestiae reprehenderit "
  }
];
