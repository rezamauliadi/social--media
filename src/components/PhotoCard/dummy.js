const albums = [
  {
    userId: 1,
    id: 1,
    title: "quidem molestiae enim"
  },
  {
    userId: 2,
    id: 11,
    title: "quidem molestiae enim"
  }
];
const photos = [
  {
    albumId: 1,
    id: 1,
    title: "accusamus beatae ad facilis cum similique qui sunt",
    url: "https://via.placeholder.com/600/92c952",
    thumbnailUrl: "https://via.placeholder.com/150/92c952"
  },
  {
    albumId: 2,
    id: 51,
    title: "non sunt voluptatem placeat consequuntur rem incidunt",
    url: "https://via.placeholder.com/600/8e973b",
    thumbnailUrl: "https://via.placeholder.com/150/8e973b"
  }
];

export default { albums, photos };
