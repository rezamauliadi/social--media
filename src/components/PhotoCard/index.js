import React from "react";
import PropTypes from "prop-types";

import { Link } from "react-router-dom";
import { Card, Image } from "semantic-ui-react";

const PhotoCard = ({ items, isAlbum }) => {
  const cardTitleStyle = {
    width: "140px",
    whiteSpace: "nowrap",
    overflow: "hidden",
    textOverflow: "ellipsis"
  };

  const cardProps = item => {
    const prop = { as: isAlbum ? Link : "a" };
    if (isAlbum) prop.to = item.url;
    else prop.href = item.url;
    return prop;
  };

  return (
    <div>
      <Card.Group itemsPerRow={3}>
        {items.map(item => (
          <Card {...cardProps(item)} key={item.id} color="blue">
            <Image
              style={{ height: "173px", width: "172px" }}
              src={item.thumbnailUrl}
              wrapped
              ui={false}
            />
            <Card.Content>
              <Card.Header style={cardTitleStyle}>{item.title}</Card.Header>
            </Card.Content>
          </Card>
        ))}
      </Card.Group>
    </div>
  );
};

PhotoCard.propTypes = {
  items: PropTypes.array
};

export default PhotoCard;
