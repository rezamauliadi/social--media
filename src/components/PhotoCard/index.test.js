import React from "react";
import { shallow } from "enzyme";

import PhotoCard from "./index.js";
import dummy from "./dummy.js";

describe("Component: PhotoCard", () => {
  it("should match snapshot when showing albums", () => {
    const props = {
      items: dummy.albums,
      isAlbum: true,
      onPhotoClick: jest.fn()
    };
    const component = shallow(<PhotoCard {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should match snapshot when showing photos", () => {
    const props = {
      items: dummy.photos,
      isAlbum: false,
      onPhotoClick: jest.fn()
    };
    const component = shallow(<PhotoCard {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should call onPhotoClick", () => {
    const fnMock = jest.fn();
    const props = {
      items: dummy.photos,
      isAlbum: false,
      onPhotoClick: fnMock
    };
    const component = shallow(<PhotoCard {...props} />);

    component
      .find("Card")
      .at(0)
      .simulate("click");

    expect(fnMock).toHaveBeenCalled();
  });
});
