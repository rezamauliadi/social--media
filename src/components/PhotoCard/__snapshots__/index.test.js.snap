// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PhotoCard should match snapshot when showing albums 1`] = `
<div>
  <CardGroup
    itemsPerRow={3}
  >
    <Card
      as={[Function]}
      color="blue"
      key="1"
    >
      <Image
        as="img"
        style={
          Object {
            "height": "173px",
            "width": "172px",
          }
        }
        ui={false}
        wrapped={true}
      />
      <CardContent>
        <CardHeader
          style={
            Object {
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
              "width": "140px",
            }
          }
        >
          quidem molestiae enim
        </CardHeader>
      </CardContent>
    </Card>
    <Card
      as={[Function]}
      color="blue"
      key="11"
    >
      <Image
        as="img"
        style={
          Object {
            "height": "173px",
            "width": "172px",
          }
        }
        ui={false}
        wrapped={true}
      />
      <CardContent>
        <CardHeader
          style={
            Object {
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
              "width": "140px",
            }
          }
        >
          quidem molestiae enim
        </CardHeader>
      </CardContent>
    </Card>
  </CardGroup>
</div>
`;

exports[`Component: PhotoCard should match snapshot when showing photos 1`] = `
<div>
  <CardGroup
    itemsPerRow={3}
  >
    <Card
      as="a"
      color="blue"
      key="1"
      onClick={[Function]}
    >
      <Image
        as="img"
        src="https://via.placeholder.com/150/92c952"
        style={
          Object {
            "height": "173px",
            "width": "172px",
          }
        }
        ui={false}
        wrapped={true}
      />
      <CardContent>
        <CardHeader
          style={
            Object {
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
              "width": "140px",
            }
          }
        >
          accusamus beatae ad facilis cum similique qui sunt
        </CardHeader>
      </CardContent>
    </Card>
    <Card
      as="a"
      color="blue"
      key="51"
      onClick={[Function]}
    >
      <Image
        as="img"
        src="https://via.placeholder.com/150/8e973b"
        style={
          Object {
            "height": "173px",
            "width": "172px",
          }
        }
        ui={false}
        wrapped={true}
      />
      <CardContent>
        <CardHeader
          style={
            Object {
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
              "width": "140px",
            }
          }
        >
          non sunt voluptatem placeat consequuntur rem incidunt
        </CardHeader>
      </CardContent>
    </Card>
  </CardGroup>
</div>
`;
